packages/compiler/test/checker/deprecation.test.ts

```diff
@@ -0,0 +1,296 @@
+import { Diagnostic } from "../../src/index.js";
+import {
+  BasicTestRunner,
+  createTestHost,
+  createTestRunner,
+  createTestWrapper,
+  expectDiagnosticEmpty,
+  expectDiagnostics,
+  TestHost,
+} from "../../src/testing/index.js";
+
+describe("compiler: checker: deprecation", () => {
+  let runner: BasicTestRunner;
+
+  beforeEach(async () => {
+    runner = await createTestRunner();
+  });
+
+  function expectDeprecations(diagnostics: readonly Diagnostic[], deprecations: string[]) {
+    expectDiagnostics(
+      diagnostics,
+      deprecations.map((x) => ({
+        code: "deprecated",
+        message: `Deprecated: ${x}`,
+        severity: "warning",
+      }))
+    );
+  }
+
+  describe("#deprecated directive", () => {
+    it("emits deprecation for use of deprecated model types", async () => {
+      const diagnostics = await runner.diagnose(`
+          #deprecated "OldFoo is deprecated"
+          model OldFoo {}
+
+          op get(): OldFoo;
+
+          model Bar {
+            foo: string | OldFoo;
+          }
+          `);
+
+      expectDeprecations(diagnostics, ["OldFoo is deprecated", "OldFoo is deprecated"]);
+    });
+
+    it("emits deprecation for use of model that extends deprecated parent", async () => {
+      const diagnostics = await runner.diagnose(`
+          #deprecated "OldFoo is deprecated"
+          model OldFoo {}
+
+          model NewFoo extends OldFoo {} // Diagnostic here
+          model IsFoo is NewFoo {} // Diagnostic here
+          model NewIsFoo is IsFoo {} // No diagnostic here
+          `);
+
+      expectDeprecations(diagnostics, ["OldFoo is deprecated", "OldFoo is deprecated"]);
```

Comment:
Just a nice to have but because a lot of the tests are making sure we get the diagnostics at the right place, could have have a logic so that you add `┆` (like extractCursor but we have many here) to the code nad it computes what position should the errors be and in expectDeprecation it compares to make sure it match

Would change the code to that
```
 #deprecated "OldFoo is deprecated"
          model OldFoo {}
          model NewFoo extends ┆OldFoo {} // Diagnostic here
          model IsFoo is ┆NewFoo {} // Diagnostic here
          model NewIsFoo is IsFoo {} // No diagnostic here
```

Reply:
I was actually looking for the code that does this but couldn't find where it gets applied to diagnostics.  I agree, it would be nice to have.

Reply:
Ahhh, I found an example, I'll add it.

Reply:
Added marker position checks